lumicks/pylake/detail/confocal.py

```diff
@@ -256,15 +263,15 @@ def _shape(self):
 
     @property
     def pixels_per_line(self):
-        return self._fast_axis_metadata["num of pixels"]
+        return self._num_pixels[self._scan_order[0]]
 
     @property
-    def _fast_axis_metadata(self):
-        return self._json["scan volume"]["scan axes"][0]
+    def _num_pixels(self):
+        return [axes["num of pixels"] for axes in self._ordered_axes()]
 
     @property
     def fast_axis(self):
-        return "X" if self._fast_axis_metadata["axis"] == 0 else "Y"
+        return "X" if self._json["scan volume"]["scan axes"][0]["axis"] == 0 else "Y"
```

Comment:
I know why this change is in this commit, but the commit message tells me that the removal of `_fast_axis_metadata` method, together with this change, should have been its own commit. Also, the way the actual variable to check the fast axis condition is extracted looks very fragile, or do we assume the json format is part of the public API? The underscore prefix in `_json` makes it look as if it is private/undocumented API that could change at any moment.

Reply:
What I didn't like was that the `_fast_axis_metadata` was used nowhere else anymore.

What's in the `_json` depends on Bluelake. It comes straight from the file. It shouldn't change, but users shouldn't depend on it, because it would lock us into maintaining a fixed internal format.

We could change it into something like:

```python
    @property
    def _scan_axes_names(self):
        axis_labels = ("X", "Y", "Z")
        return [axis_labels[ax["axis"]] for ax in self._json["scan volume"]["scan axes"]]

    @property
    def fast_axis(self):
        return self._scan_axes_names[0]
```

But Z being the fast axis never happens (and this'd be the only place where it's used), so it seems a bit superfluous (although it may communicate the meaning a bit better).

On the other hand, there is a ticket on the backlog for exposing a few more properties that would be found in the `_json` before, and the axis scanning order is one of them. I wouldn't make the API discussion for that part of this PR, so I would suggest leaving this function private until that PR.

What do you think?

Reply:
The proposed alternative doesn't address my concern, but having axis scanning order as a property would. So let's review that whenever that other PR comes around.

Reply:
Yeah, I realize that. Leave it as is for now then to not scope creep this PR?

Reply:
Yup, leave as-is. I should have been more clear on that.